cloudinit/config/cc_rpi_userdata.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (C) 2024, Raspberry Pi Ltd.
```

Comment:
Much of the module is a bit strange to me. We're using a one-shot first-boot system configuration service (cloud-init) to configure how a one-shot first-boot system configuration service (piwiz) should work. Why would people want to use both in tandem? Does piwiz do anything that cloud-init currently can't? If not, then why not just use cloud-init for everything? If no user data is passed, I can see still starting piwiz so it can be used interactively, but if a user provides cloud-init user data, I see no reason to then also use a separate configuration service.

In particular, I think allowing username and password configuration multiple ways is very confusing. Cloud-init will create a default user automatically that can be overridden to provide a different user and password. Why would then also expose additional cloud-init user data to allow piwiz to set the same thing? Let's have one configuration option for applying these things.